client-java/controller/src/main/java/org/evomaster/client/java/controller/opensearch/OpenSearchHeuristicsCalculator.java

```diff
@@ -0,0 +1,111 @@
+package org.evomaster.client.java.controller.opensearch;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.DoubleUnaryOperator;
+import org.evomaster.client.java.controller.mongo.operations.NotEqualsOperation;
+import org.evomaster.client.java.controller.opensearch.operations.ComparisonOperation;
+import org.evomaster.client.java.controller.opensearch.operations.TermOperation;
+import org.evomaster.client.java.controller.opensearch.operations.QueryOperation;
+import org.evomaster.client.java.distance.heuristics.DistanceHelper;
+import org.evomaster.client.java.sql.internal.TaintHandler;
+
+public class OpenSearchHeuristicsCalculator {
+    private final TaintHandler taintHandler;
+
+    public OpenSearchHeuristicsCalculator() {
+        this(null);
+    }
+
+    public OpenSearchHeuristicsCalculator(TaintHandler taintHandler) {
+        this.taintHandler = taintHandler;
+    }
+
+    public double computeExpression(Object query, Object doc) {
+        QueryOperation operation = getOperation(query);
+        return calculateDistance(operation, doc);
+    }
+
+    private QueryOperation getOperation(Object query) {
+        return new OpenSearchQueryParser().parse(query);
+    }
+
+    private double calculateDistance(QueryOperation operation, Object doc) {
+        if (operation instanceof TermOperation<?>) {
+            return calculateDistanceForEquals((TermOperation<?>) operation, doc);
+        }
+
+        return 0;
```

Comment:
is it correct to return 0 here? if so, it is not so obvious. add some /* */ comments to clarify it

Reply:
I guess this should be Double.MAX_VALUE as this is the case where it is not supported, right?

Reply:
Yes, will modify this.